### PR TITLE
feat(dev-portal): INFRA-194 OIDC self-service for B2C apps 😎💻 

### PR DESCRIPTION
**Pull request by SavelevMatthew**
https://github.com/open-condo-software/condo/assets/50069872/29b80aff-70b7-4cb3-bbee-68419e2e8a28

### CODE REVIEW

File: apps/dev-api/domains/miniapp/schema/OIDCClient.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        },
+        clientSecret: {
+            schemaDoc: 'OIDC Client Secret',
+            type: 'Text',
+            isRequired: true,
+            hooks: {
+                resolveInput ({ operation, originalInput, fieldPath }) {
+                    // NOTE: Secret must be strong enough, so we should generate it server-side
+                    if (operation === 'create' || originalInput.hasOwnProperty(fieldPath)) {
+                        return faker.internet.password(OIDC_SECRET_LENGTH, false, OIDC_SECRET_CHAR_PATTERN)
```
Comment (pahaz) [inline apps/dev-api/domains/miniapp/schema/OIDCClient.js:54]: it's better to use crypto! for client secret generation!

Comment (pahaz) [reply apps/dev-api/domains/miniapp/schema/OIDCClient.js:54]: SECURITY ISSUE

File: apps/dev-api/domains/miniapp/schema/PublishB2CAppService.js [hunk 2]
```diff
@@ -187,6 +207,88 @@ async function publishBuildChanges ({ build, condoBuild, app, condoApp, context,
...
+
+    let condoOIDCClient = await serverClient.findExportedModel({
+        modelGql: CondoOIDCClientGql,
+        exportId: oidcClient[exportIdField],
+        id: oidcClient.id,
+        context,
+    })
+
+    if (condoOIDCClient) {
+        logger.info({ msg:'Existing OIDC client found. Updating', appId: app.id, environment, meta: { condoOIDCClientId: condoOIDCClient.id } })
```
Comment (pahaz) [inline apps/dev-api/domains/miniapp/schema/PublishB2CAppService.js:249]:
```suggestion
        logger.info({ msg: 'Existing OIDC client found. Updating', appId: app.id, environment, meta: { condoOIDCClientId: condoOIDCClient.id } })
```

File: apps/dev-api/domains/miniapp/schema/OIDCClient.js (hunk 1 above)
Comment (SavelevMatthew) [reply apps/dev-api/domains/miniapp/schema/OIDCClient.js]: Rewritten with manual generation with `crypto` and `randomInt`. (Faker was using similar strategy btw)

File: apps/dev-api/domains/miniapp/utils/serverSchema/oidcClient.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+const crypto = require('crypto')
+
+const { OIDC_SECRET_CHAR_POOL, OIDC_SECRET_LENGTH, OIDC_CLIENT_DEFAULT_PAYLOAD } = require('@dev-api/domains/miniapp/constants/oidc')
+
+function generateClientSecret () {
+    return [...Array(OIDC_SECRET_LENGTH)]
+        .map(() => OIDC_SECRET_CHAR_POOL[crypto.randomInt(OIDC_SECRET_CHAR_POOL.length)])
+        .join('')
+}
```
Comment (sitozzz) [inline apps/dev-api/domains/miniapp/utils/serverSchema/oidcClient.js:9]: Why not just uuid v4?

**Review by sitozzz — APPROVED** [review]

**Pull request merged**